src/redux/shoppingCart.js

```diff
@@ -0,0 +1,76 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+// Helper function to save cart state to localStorage
+const saveCartToLocalStorage = (cartState) => {
+  localStorage.setItem('cartState', JSON.stringify(cartState));
+};
+
+// Helper function to load cart state from localStorage
+const loadCartFromLocalStorage = () => {
+  const savedState = localStorage.getItem('cartState');
+  return savedState ? JSON.parse(savedState) : undefined;
+};
+
+const initialState = loadCartFromLocalStorage() || {
+  items: {},
+  subtotal: 0,
+  tax: 0.15,
+  totalQuantity: 0,
+  totalAmount: 0,
+};
+
+const shoppingCartReducer = createSlice({
+  name: 'shoppingCart',
+  initialState,
+  reducers: {
+    addItem(state, action) {
+      const { album, quantity } = action.payload;
+      const albumId = album.id;
+
+      if (!state.items[albumId]) {
+        state.items[albumId] = { album, quantity: 0 };
+      }
+
+      state.items[albumId].quantity += quantity;
+      state.totalQuantity += quantity;
+      state.subtotal += album.price * quantity;
+      state.totalAmount = state.subtotal + state.subtotal * state.tax;
+
+      saveCartToLocalStorage(state);
+    },
+    reduceItem(state, action) {
+      const { album, quantity } = action.payload;
+      const albumId = album.id;
+
+      if (state.items[albumId]) {
+        const remainingQuantity = Math.max(
+          state.items[albumId].quantity - quantity,
+          0
+        );
+        state.totalQuantity -=
+          state.items[albumId].quantity - remainingQuantity;
+        state.subtotal -=
+          album.price * (state.items[albumId].quantity - remainingQuantity);
```

Comment:
Same as above comment, let's double check if this works in the case where the payload is trying to decrement more than what we have in the cart.